File: 7. Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        isMinus = False
        if x<0:
            isMinus = True
            x=-x
        x=x.__str__()
        x=x[::-1]
        x=-int(x) if isMinus else int(x)
        if x<-2147483648 or x>2147483647:
            return 0

        return x

File: 7. Reverse Integer.java

import java.util.*;


public class Solution {
    public static int reverse(int x) {
        long res = 0;
        boolean isNegative = false;
        if(x < 0){
            x *= -1;
            isNegative = true;
        }
        while(x > 0){
            res *= 10;
            res += x % 10;
            x /= 10;
        }
        if((isNegative && -res < Integer.MIN_VALUE) || (!isNegative && res > Integer.MAX_VALUE)) return 0;
        return isNegative ? (int) -res : (int) res;
    }
    public static void main(String arg[]) {
        System.out.println(reverse(321));
        System.out.println(reverse(-432));
    }
}
